src/ast/ast.cpp

```diff
@@ -241,7 +241,7 @@ bool AttachPoint::check_available(const std::string &identifier) const
 {
   ProbeType type = probetype(provider);
 
-  if (identifier == "reg" || identifier == "usermode") {
+  if (identifier == "reg" || identifier == "__usermode") {
```

Comment:
Should we standard on `__builtin_XYZ` (as does gcc, for example) for actual builtins? And provide the user API macro while we burn down the number of internal special cases?

I'm also okay with `__`, just thinking out loud.

Reply:
I like this, the `__builtin_` prefix will make easier to spot these in the code and since they are never exposed to users, it shouldn't be a problem using a longer name.

Reply:
The one downside (maybe?) of a longer prefix is in cases where there is an error and the builtin name is exposed to users via source/macro expansion. @amscanne mentioned (I can't remember where) adding some directive or something to macros to suppress expansion in errors. Anyway, I can update the prefix.